### PR TITLE
[AFR] Implemented Mordenkainen

### PR DESCRIPTION
**Pull request by weirddan455**
Issue #7808

### CODE REVIEW

File: Mage.Sets/src/mage/cards/m/Mordenkainen.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Cards hand = new CardsImpl(controller.getHand());
+        Cards library = new CardsImpl(controller.getLibrary().getCardList());
+        controller.moveCards(hand, Zone.LIBRARY, source, game);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/m/Mordenkainen.java:112]: I'm pretty sure if you're putting cards into a library you need to use the `putCardsOnTopOfLibrary` method, but it's not something I'm super sure of

File: Mage.Sets/src/mage/cards/m/Mordenkainen.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        controller.drawCards(2, source, game);
+        TargetCardInHand target = new TargetCardInHand();
+        if (controller.chooseTarget(Outcome.Discard, target, source, game)) {
+            Card card = game.getCard(target.getFirstTarget());
+            if (card != null) {
+                controller.moveCardToLibraryWithInfo(card, source, game, Zone.HAND, false, false);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/m/Mordenkainen.java:81]: I didn't even realize this method existed, feels like we shouldn't have two separate methods that do the same thing but whatever (this is fine)

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/m/Mordenkainen.java (hunk 2 above)
Comment (weirddan455) [reply Mage.Sets/src/mage/cards/m/Mordenkainen.java:81]: Yea, I'm not sure why we have two.  This is the one that Brainstorm Effect uses so I went with that.

File: Mage.Sets/src/mage/cards/m/Mordenkainen.java (hunk 1 above)
Comment (weirddan455) [reply Mage.Sets/src/mage/cards/m/Mordenkainen.java]: I changed this.  They both appear to do the same thing in game.  I guess the card ordering could be different but the library is getting shuffled immediately after anyway.

**Pull request merged**